src/Common/tests/System/Net/Http/LoopbackServer.cs

```diff
@@ -365,25 +365,167 @@ public class Options
 
         public sealed class Connection : IDisposable
         {
+            private const int _blockSize=4000;
             private Socket _socket;
             private Stream _stream;
-            private StreamReader _reader;
             private StreamWriter _writer;
+            private byte[] _buffer;
+            private int _end;
+            private int _start;
 
             public Connection(Socket socket, Stream stream)
             {
                 _socket = socket;
                 _stream = stream;
 
-                _reader = new StreamReader(stream, Encoding.ASCII);
                 _writer = new StreamWriter(stream, Encoding.ASCII) { AutoFlush = true };
+
+                _buffer = new byte[_blockSize];
+                _end= 0;
+                _start = 0;
             }
 
             public Socket Socket => _socket;
             public Stream Stream => _stream;
-            public StreamReader Reader => _reader;
             public StreamWriter Writer => _writer;
 
+            public async Task<int> ReadAsync(byte[] buffer, int offset, int size)
```

Comment:
This and a bunch of other methods here are public. Do they need to be? Seems like these should be private implementation details.

Reply:
some methods from what used to be StreamReader are used by individual tests.